cloud-minecraft/cloud-sponge/src/main/java/cloud/commandframework/sponge/argument/GameProfileCollectionArgument.java

```diff
@@ -0,0 +1,260 @@
+     *
+     * @param <C> sender type
+     */
+    public static final class Parser<C> implements NodeSupplyingArgumentParser<C, GameProfileCollection> {
+
+        private final ArgumentParser<C, GameProfileCollection> mappedParser =
+                new WrappedBrigadierParser<C, net.minecraft.commands.arguments.GameProfileArgument.Result>(
+                        net.minecraft.commands.arguments.GameProfileArgument.gameProfile()
+                ).map((ctx, argumentResult) -> {
+                    final Collection<com.mojang.authlib.GameProfile> profiles;
+                    try {
+                        profiles = argumentResult.getNames(
+                                (CommandSourceStack) ctx.get(SpongeCommandContextKeys.COMMAND_CAUSE)
+                        );
+                    } catch (final CommandSyntaxException ex) {
+                        return ArgumentParseResult.failure(ex);
+                    }
+                    final List<SpongeGameProfile> result = profiles.stream()
+                            .map(SpongeGameProfile::of).collect(Collectors.toList());
+                    return ArgumentParseResult.success(new GameProfileCollectionImpl(Collections.unmodifiableCollection(result)));
+                });
+
+        @Override
+        public @NonNull ArgumentParseResult<@NonNull GameProfileCollection> parse(
+                @NonNull final CommandContext<@NonNull C> commandContext,
+                @NonNull final Queue<@NonNull String> inputQueue
+        ) {
+            return this.mappedParser.parse(commandContext, inputQueue);
+        }
+
+        @Override
+        public @NonNull List<@NonNull String> suggestions(
+                final @NonNull CommandContext<C> commandContext,
+                final @NonNull String input
+        ) {
+            return this.mappedParser.suggestions(commandContext, input);
+        }
+
+        @Override
+        public CommandTreeNode.@NonNull Argument<? extends CommandTreeNode.Argument<?>> node() {
+            return CommandTreeNodeTypes.GAME_PROFILE.get().createNode();
+        }
+
+    }
+
+    /**
+     * Builder for {@link GameProfileCollectionArgument}.
+     *
+     * @param <C> sender type
+     */
+    public static final class Builder<C> extends TypedBuilder<C, GameProfileCollection, Builder<C>> {
+
+        Builder(final @NonNull String name) {
+            super(GameProfileCollection.class, name);
+        }
+
+        @Override
+        public @NonNull GameProfileCollectionArgument<C> build() {
+            return new GameProfileCollectionArgument<>(
+                    this.isRequired(),
+                    this.getName(),
+                    this.getDefaultValue(),
+                    this.getSuggestionsProvider(),
+                    this.getDefaultDescription()
+            );
+        }
+
+    }
+
+    @DefaultQualifier(NonNull.class)
+    private static final class GameProfileCollectionImpl implements GameProfileCollection {
```

Comment:
Could extend AbstractCollection here? Would remove the amount of methods we have to override but isn't a big deal.